src/Resources/config/config.yml

```diff
@@ -25,6 +25,13 @@ winzou_state_machine:
                     on: ['complete']
                     do: ['@sylius_invoicing_plugin.event_producer.order_payment_paid', '__invoke']
                     args: ['object']
+    sylius_order:
+        callbacks:
+            after:
+                sylius_invoicing_plugin_order_placed_producer:
+                    on: ['create']
+                    do: ['@sylius_invoicing_plugin.event_producer.order_placed', '__invoke']
+                    args: ['object']
```

Comment:
Reading what the `OrderPlacedProducer` :

- onPostPersist : when `order.checkout_state === 'completed'` then dispatch creation of `sylius_invoicing_plugin_invoice` new item
- onPostUpdate : when the new value of `order.checkout_state === 'completed'` then dispatch creation of `sylius_invoicing_plugin_invoice` new item

Don't you think the new callback should be set on the state machine named : `sylius_order_checkout` on `['complete']` ?

Reply:
I'll try this, and check if `order_number` is available at this time.
All behat tests passed for me... not for you ?

Reply:
@ehibes reproducing the change on my project I see where you had issue. The command is handle by the `InvoiceCreator` but the repository method used to retrieve the `Order` is `findOneByNumber`, which is a real repository method retrieving an `Order` with a `state` different from `cart`.

A way to solve this is to use this method instead : https://github.com/Sylius/InvoicingPlugin/blob/master/src/Doctrine/ORM/InvoiceRepositoryInterface.php#L24
This method can return multiple `Invoice`s so a `foreach` will be required then.
Finally, the same changes must be done inside `SendInvoiceEmailHandler`.

Reply:
EDIT: Better use `$order = $this->invoiceRepository->findBy(['number' => $orderNumber]);` instead.